homeassistant/components/tplink/climate.py

```diff
@@ -0,0 +1,141 @@
+"""Support for TP-Link thermostats."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from kasa import Device, DeviceType
+from kasa.smart.modules.temperaturecontrol import ThermostatState
+
+from homeassistant.components.climate import (
+    ATTR_TEMPERATURE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import PRECISION_WHOLE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .const import UNIT_MAPPING
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import CoordinatedTPLinkEntity, async_refresh_after
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TPLinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+    data = config_entry.runtime_data
+    parent_coordinator = data.parent_coordinator
+    device = parent_coordinator.device
+
+    entities: list[Climate] = []
```

Comment:
Usually we want to name entities for the integration so when we have errors or are looking for memory leaks we don't have to guess which integration a `Climate` object belongs to.  

Example 
https://www.home-assistant.io/integrations/profiler/#service-profilerdump_log_objects

Reply:
Hmm, I'm not sure I follow how I should define the entity names here? I converted this to avoid having a separate `extend` call.